# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1742898023 -3600
# Parent  878c6b10f7dc3512fa7603a585df6c92e9b24575
Bug 1952481 - Prevent message header view toolbar buttons from keeping focus after being clicked.

Based on https://phabricator.services.mozilla.com/D240727 which is overly complicated since
.querySelectorAll() already returns an iterable node list.

diff --git a/mail/base/content/msgHdrView.js b/mail/base/content/msgHdrView.js
--- a/mail/base/content/msgHdrView.js
+++ b/mail/base/content/msgHdrView.js
@@ -438,8 +438,17 @@ async function OnLoadMsgHeaderPane() {
     Ci.nsIWebProgress.NOTIFY_STATE_ALL
   );
 
   gHeaderCustomize.init();
+  for (const button of document
+    .getElementById("header-view-toolbar")
+    .querySelectorAll('.message-header-view-button:not([type="menu"])')) {
+    button.addEventListener("mousedown", event => {
+      if (event.button == 0) {
+        event.preventDefault();
+      }
+    });
+  }
 }
 
 function OnUnloadMsgHeaderPane() {
   const headerViewElement = document.getElementById("msgHeaderView");
